server/src/main/scala/com/metamx/tranquility/server/http/TranquilityServlet.scala

```diff
@@ -154,6 +157,17 @@ class TranquilityServlet(
     }
   }
 
+  private def getRequestDecompressor(): InputStream => InputStream = {
+    request.header("Content-Encoding") match {
+      case Some("gzip") | Some("x-gzip") =>
+        in => new GZIPInputStream(in)
```

Comment:
FYI: http://bugs.java.com/bugdatabase/view_bug.do?bug_id=7036144

com.metamx.common.CompressionUtils#gzipInputStream has a workaround for it